### PR TITLE
Add a wirebox:targetID injection

### PR DESCRIPTION
**Pull request by elpete**
This is used to get the targetID (usually the mapping name) used to create an object as a property on the object.

### CODE REVIEW

File: system/ioc/Builder.cfc [hunk 1]
```diff
@@ -644,7 +644,11 @@ component serializable="false" accessors="true" {
 	 * @definition The dependency definition structure: name, dsl as keys
 	 * @targetObject The target object we are building the DSL dependency for
 	 */
-	private any function getWireBoxDSL( required definition, targetObject ){
+	private any function getWireBoxDSL(
```
Comment (lmajano) [inline system/ioc/Builder.cfc:647]: I think this change needs to be consistent with every DSL get method, not only this one. Also, maybe this needs to update the `coldbox.system.ioc.dsl.IDSLBuilder` the `process()` method so we can standardize it.

Comment (lmajano) [reply system/ioc/Builder.cfc:647]: Also, can you expand on the ticket and here WHY do this?

**Review by lmajano — CHANGES_REQUESTED** [review]
Can you expand more on WHY do this? Maybe add examples here of usage?

File: system/ioc/Builder.cfc (hunk 1 above)
Comment (elpete) [reply system/ioc/Builder.cfc:647]: Sure - in Quick I have a way to get a new blank copy of the same entity.  I do this by asking WireBox for it.  But to ask WireBox for it I need to know what mapping was used.  Right now Quick does this using a `beforeInstanceAutowire`interceptor, but I thought this was cleaner.

**Comment by elpete** [comment]
Hey, @lmajano.  I answered your questions earlier above.  Anything else you need for this PR?

**Comment by lmajano** [comment]
> Hey, @lmajano. I answered your questions earlier above. Anything else you need for this PR?

Sorry, It was long forgotten. I am reviving it.

**Comment by lmajano** [comment]
Merged manually, as I needed to add more to the interface and had to do it positionally since if not, it would be a breaking change.

**Pull request closed without merging**